Pocket/pocket-ios
# feat(premium): offline alert

## Summary
* add alert for premium if a free user is offline and trying to upgrade

## References 
* [IN-1050](https://getpocket.atlassian.net/browse/IN-1050)

## Implementation Details
* added a viewmodel to handle knowledge of network and prep for storekit

## Test Steps
* make user offline
* go to settings
* go to upgrade to premium
* alert should show

## PR Checklist:

## Screenshots
![image](https://user-images.githubusercontent.com/111293209/215866682-fdb5a009-ee39-4b4a-a885-463796807635.png)


[IN-1050]: https://getpocket.atlassian.net/browse/IN-1050?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
